haystack/nodes/retriever/dense.py

```diff
@@ -115,10 +115,9 @@ def __init__(self,
 
         if devices is not None:
             self.devices = devices
-        elif use_gpu and torch.cuda.is_available():
-            self.devices = [torch.device(device) for device in range(torch.cuda.device_count())]
         else:
-            self.devices = [torch.device("cpu")]
+            device, _ = initialize_device_settings(use_cuda=use_gpu)
+            self.devices = [device]
 
```

Comment:
With this change, we would get rid of the "Multi GPU Inference capability" that was introduced in https://github.com/deepset-ai/haystack/pull/1414. 

I think it would be more meaningful if we adjusted `initialize_device_settings` to return multiple devices in the case of multiple GPUs being available. Then we can use all of the devices here in DPR and keep the multi GPU functionality. In the other nodes, we can still see if multi GPU via DataParallel is super straight-forward to implement or, if not, just fall back to single GPU and keep it for later.  

I also like the option in DPR here to pass `use_gpu=True` and optionally further restrict it to specific GPUs via `devices`. The docstring is also a bit more complete than in the other nodes. So DPR could act as a "role model" for other nodes, I think.

Reply:
I changes `initialize_device_settings` to return a list of devices instead of a single device. I looked into all nodes whether they easily allow usage of DataParallel; only the `SentenceTransformersRanker` allows to wrap its model to `DataParallel` without the need to make additional code changes. I would suggest to to add the Multi GPU inference capability for all other nodes in a separate PR.